tests/instruments/hp/test_hp8753e_with_device.py

```diff
@@ -0,0 +1,349 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2024 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import logging
+from time import sleep
+
+import numpy
+import pytest
+
+from pymeasure.adapters import PrologixAdapter
+from pymeasure.instruments.hp import HP8753E
+
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+
+@pytest.fixture(scope="module")
+def hp8753e(connected_device_address):
+    try:
+        # This allows a PrologixAdapter to be used for running the tests as well
+        #  `pytest --device-address PrologixAdapter,ASRL4::INSTR,16 -k "test_hp8753e_with_device"`
+
+        if "PrologixAdapter" not in connected_device_address:
+            vna = HP8753E(connected_device_address)
+        else:
+            _, prologix_address, gpib_address, *other_address_info = connected_device_address.split(
+                ","
+            )
+
+            prologix = PrologixAdapter(
+                resource_name=prologix_address,
+                visa_library="@py",
+                auto=1,
+                address=gpib_address,
+            )
+
+            # need to ensure `eot_enable` is set to zero otherwise you will have to read twice to
+            # get rid of the extra new line character
+            prologix.write("++eot_enable 0")
+            vna = HP8753E(adapter=prologix)
+
+    except IOError:
+        print("Not able to connect to vna")
+        assert False
+
+    yield vna
+    del vna
```

Comment:
This is an interesting solution on how to use a prologix adapter for device tests.

Reply:
This comment is for creating an additional issue.